### ISSUE TITLE
/03_data_types/06_const.md constexpr

### ISSUE THREAD

**Issue by BroBQ** [Xeverous/Xeverous.github.io#17]
I'm worried about constexpr section

`constexpr int x = __LINE__ * 2; // ok, compile-time information
`
Why do you start with macro. Quote from macros article: "There will be practically no macros used in the entire tutorial."
I wouldn't expect from reader to know `__LINE__` yet or even at all. First things first, let's start with really simple example like:
`constexpr int x = 1920 - 120;
`
I suggest this code:
```
#include <iostream>
constexpr double gravity (9.8); // ok, the value of 9.8 can be resolved at compile-time
constexpr int sum = 4 + 5; // ok, the value of 4 + 5 can be resolved at compile-time

int main()
{
    std::cout << "Enter your age: ";
    int age;
    std::cin >> age;
    // constexpr int myAge = age; // not okay, age can not be resolved at compile-time
}
```

Functions are introduced at chapter 7, so those exaples are inappropriate if we read from the beginning to the end:
```
constexpr int n1 = get_input_from_user(); // will not compile - we don't know what user will input
const int n2 = get_input_from_user(); // ok, compiler will not let change n2 but allows to initialize it from some run-time dependent value
```

**Comment by Xeverous** [comment]
> Why do you start with macro. Quote from macros article: "There will be practically no macros used in the entire tutorial."
> I wouldn't expect from reader to know `__LINE__` yet or even at all. First things first, let's start with really simple example like:

The reader will know it as it is explicitly written in macros chapter which is earlier in the tutorial.

It's obvious that such code does not make much sense - the motivation was to show that macros are compile-time information and therefore can be used in this context. I agree it should be noted that the example is only for "example" purposes that this is just possible.

A better example that have came to my mind:

    constexpr long cppstd = __cplusplus; // macro is always in the form "YYYYMM"
    constexpr long year = cppstd / 100; // remove MM part
    std::cout << "Year of the used C++ standard: " << year << "\n";

About suggested code - yes, I will add simpler examples and move the function part to functions where `constexpr` has another (somewhat different) meaning.

You criticize using functions here (hey, there is `main()` already) but use the constructor-like initilization syntax (`gravity(9.8)`). At this point, I would use `=` only for simplicity and intuitiveness.

___

1 important note: label `invalid` is not meant to tag invalid code. It is used to tag invalid issues. This issue is correct.

**Comment by Xeverous** [comment]
Fixed in 980a587d2e2a0859273eeed2667c61bfdff484b8
